internal/controllers/rest_controller.go

```diff
@@ -76,19 +140,77 @@ func (rc RESTController) OIDCProviderCallback(w http.ResponseWriter, r *http.Req
 	}
 
 	// If the flow is secure, a JWT will be created...
-	jwt, err := rc.oidcService.CreateJWT(&token.CustomClaims)
+	jwt, err := rc.oidcService.Create(&token.CustomClaims)
 	if err != nil {
 		rc.logger.Error("an error occurred while creating a JWT. Error: " + err.Error())
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	// ...and then a JSON data to display as body content.
-	jsonBody, err := json.Marshal(jwt)
-	if err != nil {
-		rc.logger.Error("an error occurred while validating some claims. Error: " + err.Error())
-		return
+	// ...set a session cookie.
+	rc.setSessionCookie(w, r, "access_token", jwt.AccessToken)
+
+	rc.logger.Infof("Redirecting back to %s", rc.configuration.AuthServiceURL)
+	http.Redirect(w, r, rc.configuration.AuthServiceURL, http.StatusFound)
+}
+
+// Logout is responsible for revoking a token and deleting an authentication cookie.
+func (rc RESTController) Logout(w http.ResponseWriter, r *http.Request) {
+	// TODO: revoke token
+	rc.deleteSessionCookie(w, "access_token")
+}
+
+// getCredentials is responsible for simply getting credentials.
+func (rc RESTController) getCredentials(r *http.Request) (string, error) {
+	rc.logger.Info("Get request credentials...")
+	for _, auth := range rc.credentials {
+		token, err := auth.Extract(r)
+		if authErrors.IsNotFound(err) {
+			continue
+		}
+		if err != nil {
+			return "", err
+		}
+		// credentials sent.
+		rc.logger.Debug(token)
+		return token, nil
 	}
+	// back to start OIDC flow.
+	return "", authErrors.ErrNotFound
+}
 
-	w.WriteHeader(http.StatusOK)
-	w.Write(jsonBody)
+func (rc RESTController) setUserInfoHeaders(w http.ResponseWriter, info models.TokenInfo) {
+	headers := map[string]string{
+		rc.configuration.UserIDHeader: rc.configuration.UserIDPrefix + info.UID,
+	}
+
+	for k, v := range headers {
+		w.Header().Set(k, v)
+	}
+}
```

Comment:
i'm pretty sure you can optimize this 🤔

Reply:
🤦 
now we just do this:
`w.Header().Set(rc.configuration.UserIDHeader, userInfo.UID)`